switch_theme.cgi

```diff
@@ -0,0 +1,47 @@
+#!/usr/local/bin/perl
+# switch_theme.cgi
+# Change the theme for the current user if allowed
+
+BEGIN { push(@INC, "."); };
+use WebminCore;
+
+&init_config();
+&ReadParse();
+my $err = sub {
+    &PrintHeader();
+    print("<tt>Cannot change theme : $_[0]</tt>\n");
+    exit(1);
+    };
+# Check if in debug mode
+&$err("Debug mode is not enabled!")
+    if (!$gconfig{'error_stack'} &&
+        !$gconfig{'debug_enabled'} &&
+        !$gconfig{'debug_theme_switcher'});
+# Check if allowed to change theme,
+# otherwise throw an error
+if (!&foreign_available('theme') &&
+    !&foreign_available('change-user') &&
+    !&foreign_available('webmin') &&
+    !&foreign_available('acl')) {
+    &$err("You are not allowed to change themes!"); 
+    }
+# Check if the theme is known
+&$err("Theme identification is not known!")
+    if (!$in{'theme'} || $in{'theme'} !~ /^[123]$/);
+# Check if the remote user is known
+&$err("Remote user is not known!") if (!$remote_user);
+# Define the theme
+my $themes = {
```

Comment:
Wouldn't it be simpler to pass in the theme directory instead of a number when switching?

Reply:
No, we need to perform strict checking.

Reply:
Do you mean preventing use of any other themes?

Reply:
> Do you mean preventing use of any other themes?

Yes. We just want to allow use of the hardcoded  themes.